design/apidsl/metadata.go

```diff
@@ -18,6 +18,11 @@ import (
 //
 //        Metadata("struct:field:name", "MyName")
 //
+// `struct:field:type`: overrides the Go struct field type generated by default by goagen.
+// Applicable to attributes only.
+//
+//        Metadata("struct:field:type", "mypackage.MyType", "github.com/me/mypackage")
```

Comment:
Just to be clear it might be good to have a first example that does not use a package. Also would be good to use a real use case for the example like the one you are after so maybe something like:
```go
//        Metadata("struct:field:type", "[]byte")
//        Metadata("struct:field:type", "json.RawMessage", "encoding/json")
```